### PR TITLE
feat(browser-extension): edit bookmark metadata after saving

### LINKED ISSUE #2882: Add title and favourites button to browser extension
### Describe the feature you'd like

Add the ability to change the title of a bookmark, also add a button to add the bookmark to the favourites list in the browser extension.

### Describe the benefits this would bring to existing Karakeep users

You wouldn't need to open the bookmark in Karakeep after creating it to change this metadata. 

### Can the goal of this request already be achieved via other means?

Yes you can open the bookmark in Karakeep to edit this info, this would just make the workflow easier when adding bookmarks with the extension.

### Have you searched for an existing open/closed issue?

- [x] I have searched for existing issues and none cover my fundamental request

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Swpn0neel**
## description

- adds an editable title to the browser extension's post-save view
- adds a favourite toggle that reflects the saved bookmark state
- preserves unsaved title text during bookmark refreshes
- prevents title and favourite updates from racing
- includes loading, retry, saved, unsaved, pending, and error states

fixes #2882

## how has this been tested?

- [x] browser-extension typecheck with typescript 5.9.3
- [x] focused oxlint check with no warnings or errors
- [x] focused oxfmt check
- [x] production vite build with 1,963 modules transformed
- [x] interface detector with no findings

the browser-extension package has no test script or component-test setup. so i did not add a new test dependency for this focused ui change.

<details><summary><h2>screenshots</h2></summary>

not included. the post-save view needs an authenticated extension session that is not available in this terminal environment.

</details>

## checklist

- [x] i have carefully read contributing.md
- [x] i have performed a self-review of my own code
- [x] documentation changes are not required for this extension ui change
- [x] i have no unrelated changes in the pull request
- [x] i have added no dependencies
- [x] the package has no existing test command, and the focused validation above passes

## llm use

codex helped research issue ownership, review edge cases and draft the PR comment. everything else was done by me.

### CODE REVIEW

File: apps/browser-extension/src/components/BookmarkMetadataEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    ) {
+      setFavouriteOverride(null);
+    }
+  }, [serverFavourite, favouriteOverride]);
+
+  const titleMutation = useUpdateBookmark({
+    onSuccess: (updatedBookmark) => {
+      const updatedTitle = updatedBookmark.title ?? "";
+      setTitleState({ draft: updatedTitle, saved: updatedTitle });
+      setTitleError(null);
```
[the remark this answers is not in the document]
Comment (Swpn0neel) [reply apps/browser-extension/src/components/BookmarkMetadataEditor.tsx:64]: fixed in e904e6b4. and the success handler now compares the current draft with the submitted title. if the user typed newer text while the request was pending, it keeps that draft and only advances the saved baseline. typecheck, lint, format, and the production extension build pass again.

**Comment by Swpn0neel** [comment]
@mohamedbassem when you have time, could you review this browser extension change? it implements the approved scope from #2882. and the typecheck, lint, format, and production build pass locally.

**Comment by keladeine** [comment]
I am really grateful. Before I inquire and poke around further, does this make the url and description editable too?

**Comment by Swpn0neel** [comment]
thanks for asking @keladeine. this pr only makes the title and favourite state editable. it does not make the url or description editable. the backend supports updating those fields, but adding them here would expand the approved scope of #2882. if the maintainers want that included, i’m happy to follow their direction. otherwise, it would be better handled as a separate issue or follow-up pr.

**Comment by keladeine** [comment]
thank you @Swpn0neel , I am sort of new at github but will figure out how to request! Thank you for being so nice and being one of my first interactions here, which intimidates me so much! 

 There's just so many freaking utm and other tracking stuff sometimes in the url and you know i want my karakeep clean and purdy. :)
